core/src/Kokkos_Core.hpp

```diff
@@ -147,34 +147,64 @@ void print_configuration(std::ostream& os, bool verbose = false);
 
 namespace Kokkos {
 
+namespace Impl {
+
+static inline void check_init_final(
+    [[maybe_unused]] const std::string& func_name) {
```

Comment:
```suggestion
    [[maybe_unused]] char const* func_name) {
```
There is no point on constructing a std::string

Reply:
Yes, this is more optimal. For a static inline function a `const char *` parameter is enough. It is within our control to pass only string literals when the function is called.

Thanks to @crtrott for pushing the change.